Healthy/Classes/Utilities/ValidatorProtocol/HealthyValidationRule .swift

```diff
@@ -0,0 +1,80 @@
+//
+//  HealthyValidationRule .swift
+//  Healthy
+//
+
+//
+
+import Foundation
+public enum HealthyValidationRule {
+
+    case matching(value: String, regex: String?)
+    case numericOnly(value: String)
+    case decimal(value: String)
+    case charCount(value: String, minCount: Int?, maxCount: Int?)
```

Comment:
I think can split this to 3 cases, minRequiredCharCount, maxRequiredCharCount, rangeRequiredCharCount to make it more flexible for the future changes.

Reply:
in this case, we have minimum  and maximum counts as an optional value so you can send them with a nil value to satisfy the 3 cases you mentioned (ex: if you want only minimum you should send the maximum value with nil )